In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
cbcl = pd.read_csv('cbcl_1_5-2023-07-21.csv')
iq = pd.read_csv("iq-2023-07-21.csv")

basic = pd.read_csv("basic_medical_screening-2023-07-21.csv")
dcdq = pd.read_csv("dcdq-2023-07-21.csv")
vineland = pd.read_csv("vineland-3-2023-07-21.csv")
rbsr = pd.read_csv("rbsr-2023-07-21.csv")
scq = pd.read_csv("scq-2023-07-21.csv")
exp = pd.read_csv("predicted_iq_experimental-2023-07-21.csv")

indi = pd.read_csv("individuals_registration-2023-07-21.csv")
background = pd.read_csv("background_history_child-2023-07-21.csv")

/tmp/ipykernel_2224318/3030006616.py:5: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.
  dcdq = pd.read_csv("dcdq-2023-07-21.csv")
/tmp/ipykernel_2224318/3030006616.py:11: DtypeWarning: Columns (34) have mixed types. Specify dtype option on import or set low_memory=False.
  indi = pd.read_csv("individuals_registration-2023-07-21.csv")


In [9]:
print("indi.shape", indi.shape)
print("exp.shape", exp.shape)
print("basic.shape", basic.shape)
print("background.shape", background.shape)
print("iq.shape", iq.shape)

indi.shape (328974, 45)
exp.shape (132368, 8)
basic.shape (192557, 133)
background.shape (47583, 117)
iq.shape (5966, 26)


In [24]:
# check no of entries in exp df with non-null values in derived_cog_impair and ml_predicted_cog_impair

print("", exp['derived_cog_impair'].value_counts(dropna=False))

print("\n", exp['ml_predicted_cog_impair'].value_counts(dropna=False))

 derived_cog_impair
1.0    57287
0.0    51514
NaN    23567
Name: count, dtype: int64

 ml_predicted_cog_impair
NaN    82747
1.0    33467
0.0    16154
Name: count, dtype: int64


|                    | derived_cog_impair | ml_predicted_cog_impair |
|--------------------|---------------------|-------------------------|
| 1.0                | 57287               | 33467                   |
| 0.0                | 51514               | 16154                   |
| NaN                | 23567               | 82747                   |

**Some entries HAVE derived_cog_impair but DO NOT HAVE ml_predicted_cog_impair .**

- This suggests that there were enough parameters to obtain the derived classification, but the predictor parameters for the model were not available.


**Some entries DO NOT HAVE derived_cog_impair but HAVE ml_predicted_cog_impair.**

- This suggests that derived_cog_impair is not a required parameter for training of model.
- There is enough data for ML prediction but not for derivation of classification? 
    - Or is the data missing due to unrelated reasons? (highly unlikely)

In [ ]:
# check which dataset contains all entries from exp

# iterate through each subject_sp_id in exp df and check for a match in indi df and basic df
# if match, do nothing. if unmatch, increase a counter
# after all iterations, print the no of entries from exp df which were not found in the reference df

unmatched_count = 0

for subject_sp_id in exp['subject_sp_id']:
    if (subject_sp_id not in indi['subject_sp_id'].values):
        unmatched_count += 1

print(unmatched_count)

In [ ]:
# check for duplicate entries in exp df